### PR TITLE
Convert to Gradle multi-module project

### PR DESCRIPTION
**Pull request by draker94**

### CODE REVIEW

File: allure-notifications-api/build.gradle [hunk 1]
```diff
@@ -0,0 +1,45 @@
+dependencies {
+    annotationProcessor(group: 'org.projectlombok', name: 'lombok', version: lombokVersion)
+    compileOnly(group: 'org.projectlombok', name: 'lombok', version: lombokVersion)
+
+    implementation platform('org.slf4j:slf4j-bom:2.0.13')
+    implementation('org.slf4j:jul-to-slf4j')   // Java Util Logging -> SLF4J
+    implementation('org.slf4j:jcl-over-slf4j') // commons-logging -> SLF4J
```
Comment (valfirst) [inline allure-notifications-api/build.gradle:7]: I think it should be move to application layer

File: allure-notifications-api/build.gradle [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    implementation(group: 'com.konghq', name: 'unirest-java', version: unirestVersion)
```
Comment (valfirst) [inline allure-notifications-api/build.gradle:9]:
```suggestion
    api(group: 'com.konghq', name: 'unirest-java', version: unirestVersion)
```

File: allure-notifications-api/build.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    implementation('org.apache.commons:commons-lang3:3.15.0')
+
+    testImplementation('org.junit.jupiter:junit-jupiter:5.10.3')
+    testRuntimeOnly('org.junit.platform:junit-platform-launcher')
+    testImplementation('org.mockito:mockito-junit-jupiter:4.11.0')
+}
+
+configurations.all {
+    exclude module: 'commons-logging' // Use jcl-over-slf4j bridge instead
+}
```
Comment (valfirst) [inline allure-notifications-api/build.gradle:23]: it should be move to application layer

File: allure-notifications-api/build.gradle [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+tasks.withType(Test).configureEach {
+    useJUnitPlatform()
+}
+
+// Checkstyle requires Java 11 starting from 10.0
+if (JavaVersion.current().isJava11Compatible()) {
```
Comment (valfirst) [inline allure-notifications-api/build.gradle:30]: it makes sense to apply it at application layer as well

File: allure-notifications-api/build.gradle
Comment (valfirst) [inline allure-notifications-api/build.gradle:1]: please add `project.description`

File: allure-notifications/build.gradle [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    compileOnly(group: 'org.projectlombok', name: 'lombok', version: lombokVersion)
+
+    implementation project(':allure-notifications-api')
+
+    implementation platform(group: 'org.apache.logging.log4j', name: 'log4j-bom', version: '2.23.1')
+    implementation(group: 'org.apache.logging.log4j', name: 'log4j-api')
+    implementation(group: 'org.apache.logging.log4j', name: 'log4j-core')
+    implementation(group: 'org.apache.logging.log4j', name: 'log4j-slf4j2-impl')
+
+    implementation(group: 'com.konghq', name: 'unirest-java', version: unirestVersion)
```
Comment (valfirst) [inline allure-notifications/build.gradle:12]:
```suggestion
```
it won't be needed if it's marked as `api` in api layer

**Review by valfirst — CHANGES_REQUESTED** [review]

File: allure-notifications/build.gradle (hunk 5 above)
Comment (draker94) [reply allure-notifications/build.gradle:12]: > it won't be needed if it's marked as `api` in api layer

Not sure why, but this dependence is not pulled up transitively (even through `api`)

**Review by valfirst — APPROVED** [review]

**Pull request merged**